python/pyarts/classes/JacobianTarget.py

```diff
@@ -0,0 +1,178 @@
+import ctypes as c
+from collections.abc import Sized
+from pyarts.workspace.api import arts_api as lib
+
+from pyarts.classes.BasicTypes import Numeric, String
+from pyarts.classes.QuantumIdentifier import QuantumIdentifier
+from pyarts.classes.io import correct_save_arguments, correct_read_arguments
+from pyarts.classes.ArrayBase import array_base
+
+
+from math import isnan
+
+class JacobianTarget:
+    """ ARTS JacobianTarget data
+
+    Properties:
+        type:
+            Enumeration type of Jacobian (String)
+
+        subtype:
+            Enumerated subtype of Jacobian (String)
+
+        perturbation:
+            Perturbation for some calculations (Numeric)
+
+        quantumidentity:
+            Identifier of the Jacobian line parameter (QuantumIdentifier)
+    """
+
+    def __init__(self, data=None):
+        if isinstance(data, c.c_void_p):
+            self.__delete__ = False
+            self.__data__ = data
+        else:
+            self.__delete__ = True
+            self.__data__ = c.c_void_p(lib.createJacobianTarget())
+            if data is not None:
+                raise RuntimeError("Only supports void initialization")
+
+    @property
+    def type(self):
+        """ Enumeration type of Jacobian (String) """
+        tmp = String(c.c_void_p(lib.enumgetTargetTypeJacobianTarget(self.__data__)))
+        tmp.__delete__ = True  # Manual delete not on by default for pointers!!!
```

Comment:
Is there a more pythonic way to deal with this @olemke ?  Like, copy-construction, but I need the __delete__ variable to no interfere with proper references...

Reply:
To avoid interfering with the internal `__delete__` attribute from outside the `String` class, it is cleaner to add an optional argument to the constructor of the `String` class. This gives the caller the option to give a hint to the `String` class how the pointer memory should be treated:
```python
class String:
    def __init__(self, value, free_memory=False):
        if isinstance(value, c.c_void_p):
            self.__delete__ = free_memory
            self.__data__ = value
        else:
            self.__delete__ = True
            self.__data__ = c.c_void_p(lib.createString())
            self.val = value
```
Adding a copy method inside the `String` class also helps with readability:
```python
    def copy(self):
        return String(str(self));
```
Then, the `type` getter in `JacobianTarget` would look like this:
```python
    @property
    def type(self):
        """ Enumeration type of Jacobian (String) """
        return String(
            c.c_void_p(lib.enumgetTargetTypeJacobianTarget(self.__data__)),
            free_memory=True,
        ).copy()
```